### PR TITLE
Todo-list project: AAA testing on Add and Delete functions

### PR DESCRIPTION
**Pull request by Yothu**
## Testing Add and Delete functions
This pull request is about testing the add and delete functions of the to-do list project, using Jest, the instructions for this project are [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md).

Files created for this pull request:
+ `__mocks__/getLC.js`
+ `tests/crud.test.js`
+ `src/getLC.js`

### Project Requirements

- [x] Use jest framework for testing.
- [x] Create a test file ([..].test.js) for a file containing the add item and delete item functions that you want to test.
- [x] If any function uses localStorage, use mock.
- [x] If any function uses DOM, mock the HTML.
- [x] Make sure you group your tests using the describe() method.

### CODE REVIEW

File: __tests__/crud.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { addTask, deleteSelectedTask } from '../src/crud-module';
+import setTasksLocalStorage from '../src/getLC';
+
+/* eslint-disable global-require */
+jest.mock('../src/getLC', () => require('../__mocks__/getLC'));
+/* eslint-enable global-require */
```
Comment (brytebee) [inline __tests__/crud.test.js]: - [x] **Required:** Do not disable the linters check. I am not familiar with using `require` but you could take a look at these resources to help you use the [ES6 syntax](https://jestjs.io/docs/manual-mocks) and [DOM manipulation.](https://jestjs.io/docs/tutorial-jquery)

File: src/index.js [hunk 2]
```diff
@@ -7,12 +7,13 @@ import modifyStatus from './status-module';
 import {
   getTasksLocalStorage,
   addTask,
-  setTasksLocalStorage,
   modifyTask,
   deleteSelectedTask,
```
Comment (brytebee) [inline src/index.js:10]: - [x] **Required:** The test should not break the functionality of your app. At the moment, the delete button/function is not working correctly. Please fix this bug.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Yothu 👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- all tests passed✔️
- correctly grouped tests using describe✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Yothu** [comment]
##### DISCLAIMER
In order to remove the linters disabler, we had to remove the `require` of the `Jest.mock` function, however, the `require` is necessary if we want the `__mocks__` folder to be outside of the `src` folder, we could not find any way to this in any other way, so we had to put the `__mocks__` folder inside the `src` folder to follow linters requirements. If this is not of the linking of the reviewer, please share with us a way to do it, because we could not find an answer even after many hours.

**Review by andersonlebon — APPROVED** [review]
## Hi @Yothu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- All tests are passing ✔️ 
## Optional suggestions
- [x] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**